src/Paystack.php

```diff
@@ -127,9 +127,23 @@ public function setResponse($relativeUrl)
         return $this;
     }
 
-    private function setGetResponse($relativeUrl)
+    private function setGetResponse($relativeUrl, $body = [])
     {
-        $this->response = $this->client->get($this->baseUrl . $relativeUrl, []);
+        $this->response = $this->client->get($this->baseUrl . $relativeUrl, $body);
+
+        return $this;
+    }
+
+    private function setPostResponse($relativeUrl, $body = [])
+    {
+        $this->response = $this->client->post($this->baseUrl . $relativeUrl, $body);
+
+        return $this;
+    }
+
+    private function setPutResponse($relativeUrl, $body = [])
+    {
+        $this->response = $this->client->put($this->baseUrl . $relativeUrl, $body);
```

Comment:
Then replace the entire code base with the setHttpResponse and pass in the relevant `get` or `post` or `put` or `delete` methods.